server/channels/app/notification.go

```diff
@@ -1147,7 +1148,7 @@ func (a *App) getGroupsAllowedForReferenceInChannel(channel *model.Channel, team
 		}
 		for _, group := range groups {
 			if group.Group.Name != nil {
```

Comment:
Is the `nil` check still relevant if we're not using it as the key? (i.e. does this code actually need to guard against "garbage groups"?)

Reply:
I think it can still be since it's omitting unnamed groups which can't be referenced. I don't entirely understand when that happens (probably something with LDAP groups), but I think it's best to leave it

Reply:
> I don't entirely understand when that happens (probably something with LDAP groups)

Not blocking this PR, but it would be awesome to make that explicit somehow.